tasks/kernel_matrix_testing/platforms.py

```diff
@@ -52,3 +55,126 @@ def filter_by_ci_component(platforms: Platforms, component: Component) -> Platfo
         new_platforms[arch] = {k: v for k, v in new_platforms[arch].items() if k in kernels}
 
     return new_platforms
+
+
+def update_image_info(ctx: Context, base_path: Path, image_info: PlatformInfo):
```

Comment:
I am wondering if we should do this in the ami-builder repository and save this information in something like a manifest file. This can then be downloaded with the image.

That way we can introduce a task to automatically generate the platform.json file based on the latest manifests when updating the agent repository, and mark this file as autogenerated.

This would mean the we do not have to download all the images in order to generate a new platforms.json. Moreover, since this file would match the state of the images in the CI, we can notify developers that they need to update their images to match the CI.

Reply:
I agree that'd be ideal, but it doesn't seem trivial. For example, we'd have to do that in the middle of the image building scripts, before compressing the images. Then we'd have to merge all the information from each job, and then think how to keep that up-to-date with the version in the repository (automated PRs?). Although doing that would also enable us to do the image versioning we discussed with less pain, it'd be part of the same process.

Reply:
Ok I re-read and understood the part about the latest manifests, that would remove the necessity to merge in the ami-builder repo, and it's easier to update locally.

Reply:
> the middle of the image building scripts, before compressing the images

Cant we use the same approach you do over here?

> Then we'd have to merge all the information from each job

We do not have to merge necessarily. The files can be upload independently similar to what we do for checksums now.

> then think how to keep that up-to-date with the version in the repository (automated PRs?)

If we implement versioning then the agent CI would continue using the version listed in platforms.json until we merge the one containing the newer versions. That process can continue to be manual.

Reply:
Hmmm. Yeah at the very least we can wait till next week to discuss it